domains/client/block-preprocessor/src/stateless_runtime.rs

```diff
@@ -349,4 +366,19 @@ where
     ) -> Result<Weight, ApiError> {
         <Self as DomainCoreApi<Block>>::extrinsic_weight(self, Default::default(), extrinsic)
     }
+
+    pub fn is_valid_sudo_call(&self, extrinsic: Vec<u8>) -> Result<bool, ApiError> {
+        <Self as DomainSudoApi<Block>>::is_valid_sudo_call(self, Default::default(), extrinsic)
+    }
+
+    pub fn construct_domain_sudo_extrinsic(
+        &self,
+        inner_call: Vec<u8>,
+    ) -> Result<Block::Extrinsic, ApiError> {
+        <Self as DomainSudoApi<Block>>::construct_domain_sudo_extrinsic(
+            self,
+            Default::default(),
+            inner_call,
+        )
+    }
```

Comment:
Should also check the domain runtime version here.

Reply:
I dont think it would make any difference since failure case already added to the client on Consensus. So we will never reach here unless Domain Runtime is upgraded

Reply:
Consensus runtime upgrade and domain runtime upgrade are 2 separate things, and even the domain runtime is upgraded, there can be FP targetting an ER that is produced before the upgrade so we still use the old runtime here.

Reply:
I know they are two seperate things @NingLin-P :)

Consensus would not allow a sudo domain call unless domain runtime is also upgraded.
So, If FP is tagetting the an ER before the runtime upgrade, FP will not include this inherent data since that specific domain_sudo_call will always fail unless consensus runtime and domain runtime are upgraded. So `generate_invalid_domain_extrinsics_root_proof` will never include the data and `construct_sudo_call` never gets called.

A malicious FP can include but it will fail at the verification since this is anyway not expected